### PR TITLE
feat(slug): callout and revert updates

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

Follow up to : https://github.com/carbon-design-system/carbon/pull/15522 & https://github.com/carbon-design-system/carbon/pull/15528

### Description

This pulls in the new callout slug updates and updates to new carbon version
### Changelog

**New**

- slug callout styles
- when slug revert is active then remove parent component background styles

**Changed**

- updated carbon version
- updated slug stories

**Removed**

- {{removed thing}}

### CODE REVIEW

File: packages/carbon-web-components/src/components/slug/slug.scss [hunk 1]
```diff
@@ -135,200 +182,217 @@
...
+:host(
+    #{$prefix}-slug:not([kind='hollow']):not(
+        [dot-type='hollow']
+      )[alignment='top-left']
+  ) {
+  .#{$prefix}--popover-caret::before {
+    border-color: $slug-callout-caret-bottom;
+    background: $slug-callout-caret-bottom-background;
   }
 }
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/slug/slug.scss:331]:
```suggestion
:host(#{$prefix}-slug:not([kind='hollow']):not([dot-type='hollow'])) {
  &[alignment='left-bottom'],
  &[alignment='right-bottom'],
  &[alignment='top'],
  &[alignment='top-right'],
  &[alignment='top-left'] {
    .#{$prefix}--popover-caret::before {
      border-color: $slug-callout-caret-bottom;
      background: $slug-callout-caret-bottom-background;
    }

    // Adjust background if actions are present
    &[has-actions] {
      .#{$prefix}--popover-caret::before {
        background: $slug-callout-caret-bottom-background-actions;
      }
    }
  }
}
```

File: packages/carbon-web-components/src/components/slug/slug.scss [hunk 2]
```diff
@@ -135,200 +182,217 @@
...
+  ),
+:host(
+    #{$prefix}-slug:not([kind='hollow']):not(
+        [dot-type='hollow']
+      )[alignment='top-left'][has-actions]
+  ) {
+  .#{$prefix}--popover-caret::before {
+    background: $slug-callout-caret-bottom-background-actions;
   }
+}
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/slug/slug.scss:362]:
```suggestion
```

I added this to the suggestion directly above. GH wouldn't let me edit both blocks at the same time.

**Review by kennylam — COMMENTED** [review]
@ariellalgilmore Looks great! Just a few suggestions for reducing repetition.

File: packages/carbon-web-components/src/components/date-picker/date-picker-input.ts [hunk 3]
```diff
@@ -332,6 +332,16 @@ class CDSDatePickerInput extends FocusMixin(LitElement) {
...
 
+  updated() {
+    this.querySelector('cds-slug')?.hasAttribute('revert-active')
+      ? this.shadowRoot
+          ?.querySelector("slot[name='slug']")
+          ?.classList.add('cds--slug--revert')
+      : this.shadowRoot
+          ?.querySelector("slot[name='slug']")
+          ?.classList.remove('cds--slug--revert');
+  }
```
[the remark this answers is not in the document]
Comment (ariellalgilmore) [reply packages/carbon-web-components/src/components/date-picker/date-picker-input.ts]: forgot about the toggle!

**Comment by ariellalgilmore** [comment]
> In the revert test, the number input is still showing a blue line under the buttons. This was a [bug](https://github.com/carbon-design-system/carbon/pull/15591) that slipped through TJs original PR as well.
> 
> image

Ahh yeah i saw his PR. We would have to wait for that PR to get added to the next release to pull in those updates

**Review by aagonzales — APPROVED** [review]
Oh ok wasn't sure if you could do anything before. It looks good other than that 👍

**Comment by ariellalgilmore** [comment]
closing in favor for: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/11453

**Pull request closed without merging**